### PR TITLE
Support filtering by resource type

### PR DESCRIPTION
**Pull request by hbuckle**
Adds a new flag `-filter` so that only the specified resource types will have tags applied.

### CODE REVIEW

**Comment by roni-frantchi** [comment]
Hey @hbuckle, thank you for the PR!  

Looks like a great feature to have in Terratag.  

Seeing that `filter` is somewhat ambiguous I think perhaps implementing it as a regex to help include/exclude resources might offer the flexibility the term suggests.  

Will you be willing to consider altering this PR to that behavior?  

And once again, thanks for contributing ❤️

**Comment by hbuckle** [comment]
@roni-frantchi sure thing, so something like this?

```
terratag -filter=azurerm_resource_group|aws_s3_bucket
```

**Comment by roni-frantchi** [comment]
Yeah @hbuckle something or even something like:  

```bash
terratag -filter='azurerm_resource_group|aws_.*'

```

File: test/fixture/terraform_15_1.0_filter/aws_azure/expected/main.terratag.tf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+provider "azurerm" {
+  features {}
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "azurerm_resource_group" "example" {
```
Comment (yaronya) [inline test/fixture/terraform_15_1.0_filter/aws_azure/expected/main.terratag.tf:21]: Could be helpful to give those resources some meaningful name per their expected status like `should_have_tags` or `should_not_have_tags`

**Review by yaronya — COMMENTED** [review]
Overall LGTM 👍 
Left you one comment about naming in your test fixture

File: test/fixture/terraform_15_1.0_filter/aws_azure/expected/main.terratag.tf (hunk 1 above)
Comment (hbuckle) [reply test/fixture/terraform_15_1.0_filter/aws_azure/expected/main.terratag.tf]: @yaronya - great call, added

**Review by roni-frantchi — APPROVED** [review]
Looking great @hbuckle!  
Thank you!

**Pull request merged**